packages/@okta/vuepress-site/docs/guides/oie-embedded-sdk-use-cases/oie-embedded-sdk-use-case-self-reg/java/integrationsteps.md

```diff
@@ -1,3 +1,251 @@
+
+### Step 7: User submits email verification code
+
+The user receives the verification code in their email and submits it in the verify code form. Use [VerifyAuthenticationOptions](https://github.com/okta/okta-idx-java/blob/master/api/src/main/java/com/okta/idx/sdk/api/model/VerifyAuthenticatorOptions.java) to capture the code and send it to the `IDXAuthenticationWrapper.verifyAuthenticator` method:
+
+```java
+VerifyAuthenticatorOptions verifyAuthenticatorOptions = new VerifyAuthenticatorOptions(code);
+AuthenticationResponse authenticationResponse =
+   idxAuthenticationWrapper.verifyAuthenticator(proceedContext, verifyAuthenticatorOptions);
+```
+
+If the request to verify the code is successful, the SDK returns an `AuthenticationResponse` object with `AuthenticationStatus=AWAITING_AUTHENTICATOR_ENROLLMENT_SELECTION` and an `Authenticators` list containing the phone factor. Reuse the authenticator enrollment form from [Step 3: Display enrollment factors](#step-3-display-enrollment-factors) to display the list of authenticators to the user.
+
+```java
+public ModelAndView selectAuthenticatorForm(AuthenticationResponse response, String title, HttpSession session) {
+   boolean canSkip = authenticationWrapper.isSkipAuthenticatorPresent(response.getProceedContext());
+   ModelAndView modelAndView = new ModelAndView("select-authenticator");
+   modelAndView.addObject("canSkip", canSkip);
+   List<String> factorMethods = new ArrayList<>();
+   for (Authenticator authenticator : response.getAuthenticators()) {
+      for (Authenticator.Factor factor : authenticator.getFactors()) {
+            factorMethods.add(factor.getMethod());
+      }
+   }
+   session.setAttribute("authenticators", response.getAuthenticators());
+   modelAndView.addObject("factorList", factorMethods);
+   modelAndView.addObject("authenticators", response.getAuthenticators());
+   modelAndView.addObject("title", title);
+   return modelAndView;
+}
+```
+
+Based on the configuration described in [Set up your Okta org for multifactor use cases](/docs/guides/oie-embedded-common-org-setup/java/main/#set-up-your-okta-org-for-multifactor-use-cases), the app in this use case is set up to require one possession factor (either email or phone). After the email factor is verified, the phone factor becomes optional. In this step, the `isSkipAuthenticatorPresent` function returns `TRUE` for the phone authenticator. You can build a **Skip** button in your form to allow the user to skip the optional phone factor. If the user decides to skip the optional factor, they are considered signed in since they have already verified the required factors. See [Step 8b, Option 2: Skip phone factor](#step-8b-option-2-skip-phone-factor) for the skip authenticator flow.
```

Comment:
Maybe update the paragraph end to something like "for the skip authenticator flow, or keep reading for the phone authenticator flow." ?

Reply:
Susan and I decided to switch the two options and changed the the step label for it to read better.